### ISSUE TITLE
[Bug]: Accessibility Issue with IconButtons

### ISSUE THREAD

**Issue by leagrdv** [storybookjs/storybook#31261]
### Describe the bug

I've identified an accessibility issue with two buttons in the sidebar of the Storybook documentation. These buttons contain only SVG icons, which makes them inaccessible to screen reader users, as there is no descriptive text or alternative labeling to convey their purpose.

To improve accessibility, it would be great if visually hidden labels could be added for these buttons. The buttons in question are:
- Clear Recently Opened Button: This button appears as a trash icon when hovering over the "Recently Opened" section.
- Clear Search Field Button: Cross icon that appears when the search input is filled. The icon is within a div which is also semantically incorrect as it should be a button.

Adding hidden labels would ensure that screen reader users can properly identify the function of these buttons.

### Reproduction link

https://design-system.post.ch/

### System

```bash
Storybook version 8.2.7
```

**Comment by Sidnioulz** [comment]
Looking at fixing this on a systemic basis by making ButtonIcon require an `aria-label`.

This has taken me through different parts of the UI and I've noticed that some ButtonIcons use a title, others a tooltip, others nothing. The title/tooltip is sometimes used to provide the action label, sometimes to provide extra details (e.g. "Highlight elements with accessibility violations" in the a11y panel).

I therefore believe ButtonIcon should have a mandatory `label` and an optional `description`
* The `label` should apply to aria-label 
* It could also apply to either tooltip or title
* The `description` should apply to aria-describedby with a `sr-hidden` DOM node
* And maybe should apply to a tooltip (instead of / additionally to the `label`?)

Dear devs, I'd like to know if you have examples of UIs where this logic would not work, for instance, cases where:
* you want the tooltip to not be a description for the button but something unrelated!?
* you don't want an aria-label D: (e.g. highlight button in a11y panel which isn't actually a ButtonIcon)
* you want a label but no title (e.g. some sidebar buttons related to search/recent item clearing)

**Comment by MichaelArestad** [comment]
@Sidnioulz

> Looking at fixing this on a systemic basis by making ButtonIcon require an aria-label.
 
I agree. I would not only like them to have an `aria-label`, but also take that same value and use it for a tooltip. 

The follow screenshot is from recent design work on Storybook's testing features, but gives you an idea of what I mean by a tooltip:

<img width="336" alt="Image" src="https://github.com/user-attachments/assets/e0c46ae9-79b4-4c9f-b607-55d8aaefc968" />

I wonder if implementing an accessible tooltip (for example a Radix primitive) could potentially solve the problem without requiring `aria-label`, but I would need to do some testing.

**Comment by Sidnioulz** [comment]
> I wonder if implementing an accessible tooltip (for example a Radix primitive) could potentially solve the problem without requiring `aria-label`, but I would need to do some testing.

Tooltips are a tricky component for accessibility compliance. They're a progressive disclosure pattern that is sometimes best mapped as screen-invisible content (e.g. you used tooltips because the layout was too dense), sometimes as an `aria-describedby` (support content only relevant to learners), sometimes as an `aria-expanded` pattern (content that might just be uninteresting to people).

Because they map into different solutions based on intent, I usually avoid baking in a specific accessibility mapping into tooltips themselves.

Instead, I'd prefer for IconButton to decide that it wants to draw a tooltip with its ARIA label. This way, other tooltips can be used with other accessibility mappings based on what fits best.

**Comment by MichaelArestad** [comment]
@Sidnioulz That all makes sense to me.

The tooltip primitive I was considering is from Radix. [Here's a short list](https://www.radix-ui.com/primitives/docs/components/tooltip#accessibility) of how it has been enhanced for accessibility. I would need to test it a bit more, but I get the impression it might be duplicative if we also used `aria-label`. It's probably simple enough to work around, but I haven't gone too deep yet.

**Comment by Sidnioulz** [comment]
I don't think Radix's tooltip is the right solution here. That component is typically useful when you want to hide content by default and have someone click a button to see it (think compact accordion). But here, we want to give users a clear understanding of what a button does when they focus it, hence `aria-label` instead.

I'll get started on a PR and maybe we can use it as a basis to discuss edge cases.

**Comment by MichaelArestad** [comment]
@Sidnioulz I think you are referring to the `popover` component which is one that shows more details. The Tooltip is designed primarily to show a short description or label for a button with no visible label. That said, I don't think a PR should be contingent on that. Thanks for getting this started!

**Comment by Sidnioulz** [comment]
https://github.com/storybookjs/storybook/discussions/23813 is relevant to this topic. At the very least, built-in tooltips in IconButton should have less display delay than a HTML `title`. I'll include it in my work list.

**Comment by Sidnioulz** [comment]
@ghengeveld @MichaelArestad I'd like to discuss how to approach hybrid Button/IconButton/Select components. Those are the ones where I see a need to have the tooltip and aria-label differ (and there are cases where Button itself could have an aria-label differing from its text content, and a tooltip on top of it).

Generally speaking, creating distinct components for Select/IconSelect and Switch/IconSwitch would make it easier to apply the right aria-markup and to provide a tooltip/aria-label API that's flexible only when it needs to be and foolproof the rest of the time.

I think it's also important that we don't treat tooltips as valid content for aria-labels (whereas we can afford to treat aria-labels as valid content for tooltips). A tooltip may rely on visual cues like the positioning of a button that are not accessible to a non-sighted person, so there will be cases where the aria-label needs to be more specific than an existing tooltip.

I've opened a PR to facilitate discussions on the use cases I've identified so far. It's not ready for merging; just opened to facilitate discussion.

**Comment by Sidnioulz** [comment]
Fixed on https://github.com/storybookjs/storybook/tree/a11y-consolidation which will be integrated into Storybook 10.
